pkg/api/unversioned/helpers.go

```diff
@@ -63,6 +63,50 @@ func LabelSelectorAsSelector(ps *LabelSelector) (labels.Selector, error) {
 	return selector, nil
 }
 
+func ParseToLabelSelector(selector string) (*LabelSelector, error) {
+	reqs, err := labels.ParseToRequirements(selector)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse the selector string \"%s\": %v", selector, err)
+	}
+
+	labelSelector := &LabelSelector{
+		MatchLabels:      map[string]string{},
+		MatchExpressions: []LabelSelectorRequirement{},
+	}
+	for _, req := range reqs {
+		var op LabelSelectorOperator
+		switch req.Operator() {
+		case labels.EqualsOperator, labels.DoubleEqualsOperator:
+			vals := req.Values()
+			if vals.Len() != 1 {
+				return nil, fmt.Errorf("equals operator must have exactly one value")
+			}
+			val, ok := vals.PopAny()
+			if !ok {
+				return nil, fmt.Errorf("equals operator has exactly one value but it cannot be retrieved")
+			}
+			labelSelector.MatchLabels[req.Key()] = val
+			continue
+		case labels.InOperator:
+			op = LabelSelectorOpIn
+		case labels.NotInOperator:
+			op = LabelSelectorOpNotIn
+		case labels.ExistsOperator:
+			op = LabelSelectorOpExists
+		case labels.DoesNotExistOperator:
+			op = LabelSelectorOpDoesNotExist
+		default:
```

Comment:
Please add an explicit case for gt and lt, so that others know it's deliberate and not an oversight.